resources/bundles/org.eclipse.core.resources/src/org/eclipse/core/internal/resources/LocationValidator.java

```diff
@@ -98,6 +98,7 @@ public IStatus validateLinkLocation(IResource resource, IPath unresolvedLocation
 		return validateLinkLocationURI(resource, URIUtil.toURI(location));
 	}
 
+	@SuppressWarnings("deprecation") // org.eclipse.core.runtime.Preferences.getBoolean(String)
```

Comment:
As you can see at https://github.com/eclipse-platform/eclipse.platform/blob/e7069db92859e15a44852ff753b9edeec2c1d7ed/resources/bundles/org.eclipse.core.resources/src/org/eclipse/core/internal/resources/PreferenceInitializer.java#L79 it's stored in DefaultScope so you can use DefaultScope.INSTANCE.getNode(ResourcesPlugin.PI_RESOURCES).getBoolean to actually move away of deprecated code.